provider/mesh/state.go

```diff
@@ -228,12 +230,17 @@ func (st *silenceState) gc(retention time.Duration) {
 	st.mtx.Lock()
 	defer st.mtx.Unlock()
 
+	keys := []uuid.UUID{}
 	t := st.now().Add(-retention)
 	for k, v := range st.m {
 		if v.EndsAt.Before(t) {
 			delete(st.m, k)
+		} else {
+			// create new key cache from existing silences
+			keys = append(keys, v.ID)
```

Comment:
Note that iteration order over the map is random. Hence the key slice here will be random as well. I think it was intended to at least be sorted by creation time?